rgdps/usecases/user_relationships.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import NamedTuple
+
+from rgdps import repositories
+from rgdps.common.context import Context
+from rgdps.constants.errors import ServiceError
+from rgdps.constants.users import UserRelationshipType
+from rgdps.models.user import User
+from rgdps.models.user_relationship import UserRelationship
+
+
+class UserRelationshipResponse(NamedTuple):
+    relationship: UserRelationship
+    user: User  # It's the user2.
+
+
+class PaginatedUserRelationshipResponse(NamedTuple):
+    relationships: list[UserRelationshipResponse]
+    total: int
+
+
+async def get_user(
+    ctx: Context,
+    user_id: int,
+    relationship_type: UserRelationshipType,
+) -> PaginatedUserRelationshipResponse | ServiceError:
+
+    relationships = await repositories.user_relationship.from_user_id(
+        ctx,
+        user_id,
+        relationship_type,
+        include_deleted=False,
+    )
+
+    relationships_responses = []
+    for relationship in relationships:
+        user = await repositories.user.from_id(ctx, relationship.user2_id)
+
+        if user is None:
+            continue
+
+        relationships_responses.append(
+            UserRelationshipResponse(relationship, user),
+        )
+
+    relationship_count = (
+        await repositories.user_relationship.get_user_relationship_count(
+            ctx,
+            user_id,
+            relationship_type,
+        )
+    )
```

Comment:
Are the extra brackets necessary?

Reply:
they are caused by black formatter, we agreed to leave it as it is in dms.